core/src/test/java/org/elasticsearch/cluster/service/ClusterServiceTests.java

```diff
@@ -382,21 +408,54 @@ public void execute() {
                     counter.incrementAndGet();
                 }
             }
+
+            @Override
+            public boolean equals(Object o) {
+                if (this == o) {
+                    return true;
+                }
+                if (o == null || getClass() != o.getClass()) {
+                    return false;
+                }
+                Task task = (Task) o;
+                return id == task.id;
+
+            }
+
+            @Override
+            public int hashCode() {
+                return id;
+            }
+
+            @Override
+            public String toString() {
+                return Integer.toString(id);
+            }
         }
 
-        int numberOfThreads = randomIntBetween(2, 8);
-        int tasksSubmittedPerThread = randomIntBetween(1, 1024);
+        int numberOfThreads = randomIntBetween(2, 5);
+        int taskSubmissionsPerThread = randomIntBetween(1, 24);
```

Comment:
I'm not sure what the average is but to me it's about maximum time. I run the original settings again (on a *laptop*) and I see runs going up to 10s, this is too long imo. Going with 8 threads and 64 tasks is still OK, so I changed to this.